spring-xd-dirt/src/main/java/org/springframework/xd/dirt/plugins/spark/streaming/MessageBusReceiver.java

```diff
@@ -107,6 +124,39 @@ public void onStop() {
 	}
 
 	/**
+	 * A {@link org.apache.spark.streaming.receiver.BlockGeneratorListener} that handles the notification
+	 * events from spark streaming {@Link BlockGenerator}.
+	 */
+	private class GeneratedBlockHandler implements BlockGeneratorListener {
+
+		public void onAddData(Object data, Object metadata) {
+			logger.debug("Adding data to block generator buffer");
+		}
+
+		public void onError(String data, Throwable t) {
+			reportError(data, t);
+		}
+
+		public void onPushBlock(StreamBlockId streamBlockId, ArrayBuffer<?> dataBuffer) {
+			store(dataBuffer);
+			LinkedList<MessageHeaders> headersListToAck = new LinkedList<MessageHeaders>();
+			for (int i = 0; i < dataBuffer.size(); i++) {
+				try {
+					headersListToAck.add(headersList.take());
+				}
+				catch (InterruptedException ie) {
```

Comment:
Whenever you catch `InterruptedException` you must invoke `Thread.currentThread().interrupt()`.